dataless-classifier/src/main/java/edu/illinois/cs/cogcomp/datalessclassification/hierarchy/SimpleTree.java

```diff
@@ -0,0 +1,338 @@
+     */
+    public boolean isRoot(N node) {
+        return _tree.isRoot(node);
+    }
+
+    /**
+     * Returns true iff <code>v1</code> is the child of <code>v2</code>. Note that if
+     * <code>v2</code> is a leaf node and <code>v1</code> is <code>null</code>, this method returns
+     * <code>true</code>.
+     */
+    public boolean isChild(N v1, N v2) {
+        return _tree.isSuccessor(v1, v2);
+    }
+
+    public boolean containsNode(N node) {
+        return _tree.containsVertex(node);
+    }
+
+    public boolean containsEdge(N v1, N v2) {
+        Integer edgeIndex = _tree.findEdge(v1, v2);
+
+        if (edgeIndex == null)
+            return false;
+        else
+            return true;
+    }
+
+    public int getNodeCount() {
+        return _tree.getVertexCount();
+    }
+
+    public Set<N> getNodes() {
+        return _tree.getVertices();
+    }
+
+    public List<N> getBreadthOrderedNodeList() {
+        List<N> output = new ArrayList<>(_tree.getVertexCount());
+        List<N> exploreNodes = new ArrayList<>();
+
+        exploreNodes.add(_tree.getRoot());
+
+        while (exploreNodes.size() != 0) {
+            N node = exploreNodes.get(0);
+            exploreNodes.remove(0);
+            output.add(node);
+
+            if (isLeaf(node) == false) {
+                Set<N> children = getChildren(node);
+                exploreNodes.addAll(children);
+            }
+        }
+
+        return output;
+    }
+
+    public List<N> getDepthOrderedNodeList() {
+        List<N> output = new ArrayList<>(_tree.getVertexCount());
+        List<N> exploreNodes = new ArrayList<>();
+
+        exploreNodes.add(_tree.getRoot());
+
+        while (exploreNodes.size() != 0) {
+            N node = exploreNodes.get(0);
+            exploreNodes.remove(0);
+            output.add(node);
+
+            if (isLeaf(node) == false) {
+                Set<N> children = getChildren(node);
+
+                for (N child : children) {
+                    exploreNodes.add(0, child);
+                }
+            }
+        }
+
+        return output;
+    }
+
+    public Set<N> getSiblingsInclusive(N node) {
+        if (_tree.containsVertex(node) == false)
+            return null;
+
+        if (isRoot(node))
+            return Collections.singleton(node);
+
+        Set<N> siblings = new HashSet<>();
+        siblings.addAll(getSiblingsExclusive(node));
+        siblings.add(node);
+
+        return new ImmutableSet.Builder<N>().addAll(siblings).build();
+    }
+
+    public Set<N> getSiblingsExclusive(N node) {
+        if (_tree.containsVertex(node) == false)
+            return null;
+
+        if (isRoot(node))
+            return null;
+
+        N parent = getParent(node);
+
+        Set<N> siblings = new HashSet<>();
+
+        siblings.addAll(getChildren(parent));
+
+        siblings.remove(node);
+
+        return new ImmutableSet.Builder<N>().addAll(siblings).build();
+    }
+
+    public List<N> getAllParents(N node) {
```

Comment:
We used the name Tree, because we expect the end-user to only provide a Tree-like label hierarchy. As for our underlying implementation, we use a full-fledged graph library since our Dataless Classifier might start incorporating relations and different weights to different edges in the future, plus it makes the functions more efficient and easier to implement, with very little additional memory consumption.